### PR TITLE
Add initial components for web app frontend

### PR DESCRIPTION
**Pull request by samirpdl**
- Added new components
- Removed external CSS and JS

### CODE REVIEW

File: frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import IP from './IP';
+import Type from './Type';
+import Heading from './Heading';
+
+export default class CardHeader extends React.Component {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Index.js]: Use function or arrow function to create pure UI / markup only components. 

Something like this:
```jsx

const CardHeader = () => (
  <div className="card-header bg-secondary border-secondary clearfix">
    ...
  </div>
);

export default CardHeader;
```

File: frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Type.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+export default class Type extends React.Component {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Type.js]: .

File: frontend/src/components/GridView/Project/Card/Card-Item/Description/Description-Item.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React, { PureComponent } from 'react'
+
+export default class DescriptionItem extends PureComponent {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/Description/Description-Item.js]: move `export default` separately to the bottomm

File: frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Type.js (hunk 2 above)
Comment (samirpdl) [reply frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Type.js]: Sure. Will change on all.

File: frontend/src/components/GridView/Project/Card/Card-Item/Service/Index.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React, { PureComponent } from 'react';
+
+import ServiceItem from './Service-Item';
+
+export default class Service extends PureComponent {
+    render() {
+        return (
+            <div className="card-body text-secondary">
+                {this.props.services.map(function (service, i) {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/Service/Index.js]: Use arrow functions for all the anonymous functions or callbacks to be passed. Only use `function` keyword for root level functions that must have a name.

File: frontend/src/components/GridView/Project/Card/Index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import CardItem from './Card-Item/Index'
+import EnvironmentHeading from './Environment-Heading';
+
+export default class Card extends React.Component {
+    render() {
+        return (<section className="card-wrapper">
+            <EnvironmentHeading title={this.props.environment.title} />
+            <div className="d-flex justify-content-left flex-wrap">
+                {this.props.environment.servers.map(function (server, j) {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Index.js]: arrow function

File: frontend/src/components/GridView/Project/Index.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React, { PureComponent } from 'react'
+
+import Card from './Card/Index';
+
+export default class Project extends PureComponent {
+    render() {
+        return (<section>
+            <h1>{this.props.project.project}</h1>
+            {this.props.project.environments.map(function (environment, i) {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Index.js]: same here

File: frontend/src/components/TableView/Index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React, { PureComponent } from 'react'
+// import Project from '../GridView/Project'
+
+export default class TableView extends PureComponent {
+    render() {
+        return (
+            <table className="table table-striped">
+                {/* {this.props.projects.map(function (project) {
```
Comment (kabirbaidhya) [inline frontend/src/components/TableView/Index.js]: Remove unnecessary comments

File: frontend/src/containers/index.css [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  -moz-osx-font-smoothing: grayscale;
+}
+
+code {
+  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
+    monospace;
+}
+
+*,::after,::before{box-sizing:border-box;}
+main,section{display:block;}
```
Comment (kabirbaidhya) [inline frontend/src/containers/index.css]: Need proper code formatting for this file. Need to use a linter / prettier

File: frontend/src/components/GridView/Index.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import React, { PureComponent } from 'react'
+
+import Project from './Project/Index';
+
+export default class GridView extends PureComponent {
+    render() {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Index.js]: Inconsistent indentation. Use 2 spaces everywhere across the codebase

File: frontend/src/components/GridView/Index.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+        return (
+            <section>
+                {this.props.projects.map(function (project, i) {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Index.js]: arrow function

File: frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/IP.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+export default class IP extends React.Component {
+    render() {
+        return (
+            <span className="ip-address">{this.props.ip}</span>
+        )
+    }
+}
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/IP.js]: strictly define prop types for the components that has props

File: README.md [hunk 12]
```diff
@@ -1 +1,8 @@
-# Koothrappali 
+Koothrapali
+
+1. yarn install
```
Comment (kabirbaidhya) [inline README.md]: Use code block formatting in markdown like this:
```
$ yarn install
$ cp .env.example .env
$ yarn start
```

File: frontend/src/components/GridView/Project/Card/Environment-Heading.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+export default class EnvironmentHeading extends React.Component {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Environment-Heading.js]: One line above this

File: frontend/src/components/GridView/Project/Card/Card-Item/Description/Index.js [hunk 14]
```diff
@@ -0,0 +1,16 @@
+import React, { PureComponent } from 'react'
+
+import DescriptionItem from './Description-Item';
+
+export default class Description extends PureComponent {
+    render() {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/Description/Index.js]:
```js
const { details } = this.props;
```

File: frontend/src/components/TableView/Index.js (hunk 7 above)
Comment (kabirbaidhya) [inline frontend/src/components/TableView/Index.js]: .

**Comment by samirpdl** [comment]
@kabirbaidhya Solved all issues.

File: frontend/src/components/GridView/Index.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Project from './Project/Index';
+
+const GridView = (props) => (
+  <section>
+    {props.projects.map((project, i) => {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Index.js]: Braces can be omitted where ever possible. 
```js
{props.projects.map((project, i) => (
  <Project project={project} key={i} />
))}
```

File: frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Index.js [hunk 16]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import IP from './IP';
+import Type from './Type';
+import Heading from './Heading';
+
+const CardHeader = (props) => {
+  const { server } = props;
+  return (
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/CardHeader/Index.js]: One blank line in between dissimilar blocks.

```js
const { server } = props;

return (...);
```

File: frontend/src/components/GridView/Project/Card/Card-Item/Description/Index.js [hunk 17]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+
+import DescriptionItem from './Description-Item';
+
+const Description = (props) => {
+  const { details } = props;
+  return (
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/Description/Index.js]: one blank line

File: frontend/src/components/GridView/Project/Card/Card-Item/Service/Index.js [hunk 18]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ServiceItem from './Service-Item';
+
+const Service = (props) => (
+  <div className="card-body text-secondary">
+    {props.services.map((service, i) => {
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Card-Item/Service/Index.js]: return and braces could be omitted

File: frontend/src/components/GridView/Project/Card/Index.js [hunk 19]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import CardItem from './Card-Item/Index'
+import EnvironmentHeading from './Environment-Heading';
+
+const Card = (props) => {
+  const { environment } = props;
+  return (
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Index.js]: one blank line

File: frontend/src/components/GridView/Project/Card/Index.js [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+    <section className="card-wrapper">
+      <EnvironmentHeading title={environment.title} />
+      <div className="d-flex justify-content-left flex-wrap">
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Card/Index.js]: omit return and braces

File: frontend/src/components/GridView/Project/Index.js [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+
+import Card from './Card/Index';
+
+const Project = (props) => {
+  const { project } = props;
+  return (
+    <section>
+      <h1>{project.project}</h1>
+      {project.environments.map((environment, i) => {
+        return <Card environment={environment} key={i} />
```
Comment (kabirbaidhya) [inline frontend/src/components/GridView/Project/Index.js]: .

File: frontend/src/components/TableView/Index.js [hunk 22]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+
+const TableView = () => (
+    <table className="table table-striped">
```
Comment (kabirbaidhya) [inline frontend/src/components/TableView/Index.js]: indentation

File: frontend/src/index.css [hunk 23]
```diff
@@ -1,14 +1,340 @@
...
 code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
-    monospace;
+    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
 }
+
+*,
+::after,
+::before {
+    box-sizing: border-box;
```
Comment (kabirbaidhya) [inline frontend/src/index.css]: indentation should be consistent everywhere. 2 spaces.

File: frontend/src/App.js [hunk 24]
```diff
@@ -1,26 +1,47 @@
...
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
-      </div>
+      <main role="main" className="container-fluid">
+        <div className="btn-group float-right">
+          <a href="#grid-view" className={this.state.isGridView ? 'btn btn-success' : 'btn btn-primary'} onClick={this.showGridView}>Grid View</a>
```
Comment (kabirbaidhya) [inline frontend/src/App.js]: Can destructure these data for cleanliness.
```js
const { isGridView, data } = this.state
```

**Comment by samirpdl** [comment]
@kabirbaidhya Improved.

File: frontend/src/utils/http.js [hunk 25]
```diff
@@ -0,0 +1,9 @@
+import 'whatwg-fetch'
+
+export function get(url, raw = false) {
```
Comment (kabirbaidhya) [inline frontend/src/utils/http.js]: doc blocks

Comment (kabirbaidhya) [reply frontend/src/utils/http.js]: Also, please note that every root level function needs to have a doc block.

File: frontend/src/App.js [hunk 26]
```diff
@@ -1,26 +1,49 @@
...
+
+    this.state = {
+      data: null,
+      isGridView: true
+    };
+  }
+
+  componentDidMount() {
+    http.get(config.apiBaseUrl)
+      .then(response => response.json())
```
Comment (kabirbaidhya) [inline frontend/src/App.js]: I don't think this line is necessary now, since we have done this in the `http` util itself

**Pull request merged**